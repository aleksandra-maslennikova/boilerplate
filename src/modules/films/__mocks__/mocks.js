export const filmsMock = {
  count: 6,
  results: [
    {
      title: "A New Hope",
      episode_id: 4,
      opening_crawl: "It is a period of civil war",
      director: "George Lucas",
      producer: "Gary Kurtz, Rick McCallum",
      release_date: "1977-05-25",
      characters: ["http://swapi.dev/api/people/1/"],
      planets: ["http://swapi.dev/api/planets/1/"],
      starships: [],
      vehicles: [],
      species: [],
      created: "2014-12-10T14:23:31.880000Z",
      edited: "2014-12-20T19:49:45.256000Z",
      url: "http://swapi.dev/api/films/1/",
    },
  ],
};

export const filmMock = {
  title: "A New Hope",
  episode_id: 4,
  opening_crawl: "It is a period of civil war",
  director: "George Lucas",
  producer: "Gary Kurtz, Rick McCallum",
  release_date: "1977-05-25",
  characters: ["http://swapi.dev/api/people/1/"],
  planets: ["http://swapi.dev/api/planets/1/"],
  starships: [],
  vehicles: [],
  species: [],
  created: "2014-12-10T14:23:31.880000Z",
  edited: "2014-12-20T19:49:45.256000Z",
  url: "http://swapi.dev/api/films/1/",
};
