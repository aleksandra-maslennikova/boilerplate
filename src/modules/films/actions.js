/* Api */
import { api } from "../../api";

/* Types */
import { types } from "./types";

export const filmsActions = Object.freeze({
  /* Sync */
  startFetching: () => {
    return {
      type: types.START_FETCHING_FILMS,
    };
  },
  stopFetching: () => {
    return {
      type: types.STOP_FETCHING_FILMS,
    };
  },
  fill: (payload) => {
    return {
      type: types.FILL_FILMS,
      payload,
    };
  },
  setFetchingError: (error) => {
    return {
      type: types.SET_FETCHING_FILMS_ERROR,
      error: true,
      payload: error,
    };
  },

  /* Async */
  fetchAsync: (page) => async (dispatch) => {
    dispatch(filmsActions.startFetching());

    const response = await api.films.fetch(page);

    if (response.status === 200) {
      const { results, count } = await response.json();

      dispatch(filmsActions.fill({ results, count }));
    } else {
      const error = {
        status: response.status,
      };

      dispatch(filmsActions.setFetchingError(error));
    }
    dispatch(filmsActions.stopFetching());
  },
});

export const filmActions = Object.freeze({
  /* Sync */
  startFetching: () => {
    return {
      type: types.START_FETCHING_FILM,
    };
  },
  stopFetching: () => {
    return {
      type: types.STOP_FETCHING_FILM,
    };
  },
  fillFilmProfile: (payload) => {
    return {
      type: types.FILL_FILM_PROFILE,
      payload,
    };
  },
  setFetchingError: (error) => {
    return {
      type: types.SET_FETCHING_FILM_ERROR,
      error: true,
      payload: error,
    };
  },
  clearFilmProfile: () => {
    return {
      type: types.CLEAR_FILM_PROFILE,
    };
  },

  /* Async */
  fetchAsync: (id) => async (dispatch) => {
    dispatch(filmActions.startFetching());

    const response = await api.film.fetch(id);

    if (response.status === 200) {
      const results = await response.json();

      dispatch(filmActions.fillFilmProfile(results));
    } else {
      const error = {
        status: response.status,
      };

      dispatch(filmActions.setFetchingError(error));
    }
    dispatch(filmActions.stopFetching());
  },
});
