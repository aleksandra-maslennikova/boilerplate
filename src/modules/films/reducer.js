/* Types */
import { types } from "./types";

const initialState = {
  /* Films */
  data: [],
  count: 0,
  isFetching: false,
  error: null,

  /* Film */
  film: null,
  isFetchingFilm: false,
  filmError: null,
};

export const filmsReducer = (state = initialState, { type, payload }) => {
  switch (type) {
    /* Films */
    case types.START_FETCHING_FILMS:
      return {
        ...state,
        isFetching: true,
      };
    case types.STOP_FETCHING_FILMS:
      return {
        ...state,
        isFetching: false,
      };
    case types.SET_FETCHING_FILMS_ERROR:
      return {
        ...state,
        error: payload,
        data: [],
        count: 0,
      };
    case types.FILL_FILMS:
      return {
        ...state,
        data: payload.results.map((item, index) => ({
          ...item,
          id: index + 1,
        })),
        count: payload.count,
        error: null,
      };

    /* Film */
    case types.START_FETCHING_FILM:
      return {
        ...state,
        isFetchingFilm: true,
      };
    case types.STOP_FETCHING_FILM:
      return {
        ...state,
        isFetchingFilm: false,
      };
    case types.SET_FETCHING_FILM_ERROR:
      return {
        ...state,
        filmError: payload,
        film: null,
      };
    case types.FILL_FILM_PROFILE:
      return {
        ...state,
        film: payload,
        filmError: null,
      };
    case types.CLEAR_FILM_PROFILE:
      return {
        ...state,
        film: null,
        filmError: null,
      };

    default:
      return state;
  }
};
