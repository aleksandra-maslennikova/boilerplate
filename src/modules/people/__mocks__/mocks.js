export const peopleMock = {
  count: 82,
  results: [
    {
      name: "Wilhuff Tarkin",
      height: "180",
      mass: "unknown",
      hair_color: "auburn, grey",
      skin_color: "fair",
      eye_color: "blue",
      birth_year: "64BBY",
      gender: "male",
      homeworld: "http://swapi.dev/api/planets/21/",
      films: ["http://swapi.dev/api/films/1/", "http://swapi.dev/api/films/6/"],
      species: [],
      vehicles: [],
      starships: [],
      created: "2014-12-10T16:26:56.138000Z",
      edited: "2014-12-20T21:17:50.330000Z",
      url: "http://swapi.dev/api/people/12/",
    },
  ],
};

export const personMock = {
  name: "Luke Skywalker",
  height: "172",
  mass: "77",
  hair_color: "blond",
  skin_color: "fair",
  eye_color: "blue",
  birth_year: "19BBY",
  gender: "male",
  homeworld: "http://swapi.dev/api/planets/1/",
  films: [
    "http://swapi.dev/api/films/1/",
    "http://swapi.dev/api/films/2/",
    "http://swapi.dev/api/films/3/",
    "http://swapi.dev/api/films/6/",
  ],
  species: [],
  vehicles: [
    "http://swapi.dev/api/vehicles/14/",
    "http://swapi.dev/api/vehicles/30/",
  ],
  starships: [
    "http://swapi.dev/api/starships/12/",
    "http://swapi.dev/api/starships/22/",
  ],
  created: "2014-12-09T13:50:51.644000Z",
  edited: "2014-12-20T21:17:56.891000Z",
  url: "http://swapi.dev/api/people/1/",
};
