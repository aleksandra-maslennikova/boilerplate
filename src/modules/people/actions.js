/* Api */
import { api } from "../../api";

/* Types */
import { types } from "./types";

export const peopleActions = Object.freeze({
  /* Sync */
  startFetching: () => {
    return {
      type: types.START_FETCHING_PEOPLE,
    };
  },
  stopFetching: () => {
    return {
      type: types.STOP_FETCHING_PEOPLE,
    };
  },
  fill: (payload) => {
    return {
      type: types.FILL_PEOPLE,
      payload,
    };
  },
  setFetchingError: (error) => {
    return {
      type: types.SET_FETCHING_PEOPLE_ERROR,
      error: true,
      payload: error,
    };
  },

  /* Async */
  fetchAsync: (page) => async (dispatch) => {
    dispatch(peopleActions.startFetching());

    const response = await api.people.fetch(page);

    if (response.status === 200) {
      const { results, count } = await response.json();

      dispatch(peopleActions.fill({ results, count }));
    } else {
      const error = {
        status: response.status,
      };

      dispatch(peopleActions.setFetchingError(error));
    }
    dispatch(peopleActions.stopFetching());
  },
});

export const personActions = Object.freeze({
  /* Sync */
  startFetching: () => {
    return {
      type: types.START_FETCHING_PERSON,
    };
  },
  stopFetching: () => {
    return {
      type: types.STOP_FETCHING_PERSON,
    };
  },
  fillPersonProfile: (payload) => {
    return {
      type: types.FILL_PERSON_PROFILE,
      payload,
    };
  },
  setFetchingError: (error) => {
    return {
      type: types.SET_FETCHING_PERSON_ERROR,
      error: true,
      payload: error,
    };
  },
  clearPersonProfile: () => {
    return {
      type: types.CLEAR_PERSON_PROFILE,
    };
  },

  /* Async */
  fetchAsync: (id) => async (dispatch) => {
    dispatch(personActions.startFetching());

    const response = await api.person.fetch(id);

    if (response.status === 200) {
      const results = await response.json();

      dispatch(personActions.fillPersonProfile(results));
    } else {
      const error = {
        status: response.status,
      };

      dispatch(personActions.setFetchingError(error));
    }
    dispatch(personActions.stopFetching());
  },
});
