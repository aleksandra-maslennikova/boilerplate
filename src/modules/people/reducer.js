/* Types */
import { types } from "./types";

const initialState = {
  /* People */
  data: [],
  count: 0,
  isFetching: false,
  error: null,

  /* Person */
  person: null,
  isFetchingPerson: false,
  personError: null,
};

export const peopleReducer = (state = initialState, { type, payload }) => {
  switch (type) {
    /* People */
    case types.START_FETCHING_PEOPLE:
      return {
        ...state,
        isFetching: true,
      };
    case types.STOP_FETCHING_PEOPLE:
      return {
        ...state,
        isFetching: false,
      };
    case types.SET_FETCHING_PEOPLE_ERROR:
      return {
        ...state,
        error: payload,
        data: [],
        count: 0,
      };
    case types.FILL_PEOPLE:
      return {
        ...state,
        data: payload.results.map((item, index) => ({
          ...item,
          id: index + 1,
        })),
        count: payload.count,
        error: null,
      };

    /* Person */
    case types.START_FETCHING_PERSON:
      return {
        ...state,
        isFetchingPerson: true,
      };
    case types.STOP_FETCHING_PERSON:
      return {
        ...state,
        isFetchingPerson: false,
      };
    case types.SET_FETCHING_PERSON_ERROR:
      return {
        ...state,
        personError: payload,
        person: null,
      };
    case types.FILL_PERSON_PROFILE:
      return {
        ...state,
        person: payload,
        personError: null,
      };
    case types.CLEAR_PERSON_PROFILE:
      return {
        ...state,
        person: null,
        personError: null,
      };

    default:
      return state;
  }
};
