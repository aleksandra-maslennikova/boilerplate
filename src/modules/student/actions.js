import { types } from "./types";

export const studentActions = {
  fillProfile: (profile) => {
    return {
      type: types.FILL_STUDENT_PROFILE,
      payload: profile,
    };
  },
};
