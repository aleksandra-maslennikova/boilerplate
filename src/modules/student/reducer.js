import { types } from "./types";

const initialState = {
  data: {
    firstName: "",
    surname: "",
    email: "",
    age: "",
    specialty: "",
    sex: "",
  },
};

export const studentReducer = (state = initialState, action) => {
  switch (action.type) {
    case types.FILL_STUDENT_PROFILE: {
      return {
        ...state,
        data: action.payload,
      };
    }

    default:
      return state;
  }
};
