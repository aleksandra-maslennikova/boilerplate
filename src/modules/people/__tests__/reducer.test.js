/* Reducer */
import { peopleReducer } from "../reducer";

/* Actions */
import { peopleActions, personActions } from "../actions";

/* Mocks */
import { peopleMock, personMock } from "../__mocks__/mocks";

describe("People reducer people actions", () => {
  test("should return initial state by default", () => {
    expect(peopleReducer(void 0, {})).toMatchSnapshot();
  });
  test("should handle start fetching action", () => {
    expect(
      peopleReducer(void 0, peopleActions.startFetching())
    ).toMatchSnapshot();
  });
  test("should handle stop fetching action", () => {
    expect(
      peopleReducer(void 0, peopleActions.stopFetching())
    ).toMatchSnapshot();
  });
  test("should handle fill action", () => {
    expect(
      peopleReducer(void 0, peopleActions.fill(peopleMock))
    ).toMatchSnapshot();
  });
  test("should handle set fetching error message", () => {
    expect(
      peopleReducer(void 0, peopleActions.setFetchingError("We have an error"))
    ).toMatchSnapshot();
  });
});

describe("People reducer person actions", () => {
  test("should return initial state by default", () => {
    expect(peopleReducer(void 0, {})).toMatchSnapshot();
  });
  test("should handle start fetching action", () => {
    expect(
      peopleReducer(void 0, personActions.startFetching())
    ).toMatchSnapshot();
  });
  test("should handle stop fetching action", () => {
    expect(
      peopleReducer(void 0, personActions.stopFetching())
    ).toMatchSnapshot();
  });
  test("should handle fill action", () => {
    expect(
      peopleReducer(void 0, personActions.fillPersonProfile(personMock))
    ).toMatchSnapshot();
  });
  test("should handle set fetching error message", () => {
    expect(
      peopleReducer(void 0, personActions.setFetchingError("We have an error"))
    ).toMatchSnapshot();
  });
  test("should handle clear action", () => {
    expect(
      peopleReducer(void 0, personActions.clearPersonProfile())
    ).toMatchSnapshot();
  });
});
