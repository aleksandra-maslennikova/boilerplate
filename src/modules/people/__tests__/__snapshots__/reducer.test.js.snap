// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`People reducer people actions should handle fill action 1`] = `
Object {
  "count": 82,
  "data": Array [
    Object {
      "birth_year": "64BBY",
      "created": "2014-12-10T16:26:56.138000Z",
      "edited": "2014-12-20T21:17:50.330000Z",
      "eye_color": "blue",
      "films": Array [
        "http://swapi.dev/api/films/1/",
        "http://swapi.dev/api/films/6/",
      ],
      "gender": "male",
      "hair_color": "auburn, grey",
      "height": "180",
      "homeworld": "http://swapi.dev/api/planets/21/",
      "id": 1,
      "mass": "unknown",
      "name": "Wilhuff Tarkin",
      "skin_color": "fair",
      "species": Array [],
      "starships": Array [],
      "url": "http://swapi.dev/api/people/12/",
      "vehicles": Array [],
    },
  ],
  "error": null,
  "isFetching": false,
  "isFetchingPerson": false,
  "person": null,
  "personError": null,
}
`;

exports[`People reducer people actions should handle set fetching error message 1`] = `
Object {
  "count": 0,
  "data": Array [],
  "error": "We have an error",
  "isFetching": false,
  "isFetchingPerson": false,
  "person": null,
  "personError": null,
}
`;

exports[`People reducer people actions should handle start fetching action 1`] = `
Object {
  "count": 0,
  "data": Array [],
  "error": null,
  "isFetching": true,
  "isFetchingPerson": false,
  "person": null,
  "personError": null,
}
`;

exports[`People reducer people actions should handle stop fetching action 1`] = `
Object {
  "count": 0,
  "data": Array [],
  "error": null,
  "isFetching": false,
  "isFetchingPerson": false,
  "person": null,
  "personError": null,
}
`;

exports[`People reducer people actions should return initial state by default 1`] = `
Object {
  "count": 0,
  "data": Array [],
  "error": null,
  "isFetching": false,
  "isFetchingPerson": false,
  "person": null,
  "personError": null,
}
`;

exports[`People reducer person actions should handle clear action 1`] = `
Object {
  "count": 0,
  "data": Array [],
  "error": null,
  "isFetching": false,
  "isFetchingPerson": false,
  "person": null,
  "personError": null,
}
`;

exports[`People reducer person actions should handle fill action 1`] = `
Object {
  "count": 0,
  "data": Array [],
  "error": null,
  "isFetching": false,
  "isFetchingPerson": false,
  "person": Object {
    "birth_year": "19BBY",
    "created": "2014-12-09T13:50:51.644000Z",
    "edited": "2014-12-20T21:17:56.891000Z",
    "eye_color": "blue",
    "films": Array [
      "http://swapi.dev/api/films/1/",
      "http://swapi.dev/api/films/2/",
      "http://swapi.dev/api/films/3/",
      "http://swapi.dev/api/films/6/",
    ],
    "gender": "male",
    "hair_color": "blond",
    "height": "172",
    "homeworld": "http://swapi.dev/api/planets/1/",
    "mass": "77",
    "name": "Luke Skywalker",
    "skin_color": "fair",
    "species": Array [],
    "starships": Array [
      "http://swapi.dev/api/starships/12/",
      "http://swapi.dev/api/starships/22/",
    ],
    "url": "http://swapi.dev/api/people/1/",
    "vehicles": Array [
      "http://swapi.dev/api/vehicles/14/",
      "http://swapi.dev/api/vehicles/30/",
    ],
  },
  "personError": null,
}
`;

exports[`People reducer person actions should handle set fetching error message 1`] = `
Object {
  "count": 0,
  "data": Array [],
  "error": null,
  "isFetching": false,
  "isFetchingPerson": false,
  "person": null,
  "personError": "We have an error",
}
`;

exports[`People reducer person actions should handle start fetching action 1`] = `
Object {
  "count": 0,
  "data": Array [],
  "error": null,
  "isFetching": false,
  "isFetchingPerson": true,
  "person": null,
  "personError": null,
}
`;

exports[`People reducer person actions should handle stop fetching action 1`] = `
Object {
  "count": 0,
  "data": Array [],
  "error": null,
  "isFetching": false,
  "isFetchingPerson": false,
  "person": null,
  "personError": null,
}
`;

exports[`People reducer person actions should return initial state by default 1`] = `
Object {
  "count": 0,
  "data": Array [],
  "error": null,
  "isFetching": false,
  "isFetchingPerson": false,
  "person": null,
  "personError": null,
}
`;
